File: internal/file/block_id_test.go
package file_test

import (
	"testing"

	"kura/internal/file"
)

func TestBlockID(t *testing.T) {
	// Arrange
	filename := "testfile"
	blockNum := int64(123)
	blockID := file.NewBlockID(filename, blockNum)

	// Act/Assert
	// Test Filename method
	if blockID.Filename() != filename {
		t.Errorf("Filename() = %s; want %s", blockID.Filename(), filename)
	}

	// Test Number method
	if blockID.Number() != blockNum {
		t.Errorf("Number() = %d; want %d", blockID.Number(), blockNum)
	}

	// Test String method
	expectedString := "[file testfile, block 123]"
	if blockID.String() != expectedString {
		t.Errorf("String() = %s; want %s", blockID.String(), expectedString)
	}

	// Test Hash method
	hash1 := blockID.Hash()
	hash2 := blockID.Hash()
	if hash1 != hash2 {
		t.Errorf("Hash() is not consistent; got %d and %d", hash1, hash2)
	}
}

File: app/src/test/kotlin/com/kura/file/BlockIdTest.kt
package com.kura.file

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class BlockIdTest {
    @Test
    fun `test BlockId properties and methods`() {
        // Arrange
        val fileName = "testfile"
        val blockNum = 123
        val blockId = BlockId(fileName, blockNum)

        // Assert
        assertEquals(fileName, blockId.fileName, "fileName should match the constructor argument")
        assertEquals(blockNum, blockId.blockNum, "number should match the constructor argument")
        assertEquals("[file testfile, block 123]", blockId.toString(), "toString should return the correct format")

        // Test hash consistency
        val hash1 = blockId.hashCode()
        val hash2 = blockId.hashCode()
        assertEquals(hash1, hash2, "hashCode should be consistent")

        // Test data class equality
        val sameBlockId = BlockId(fileName, blockNum)
        assertEquals(blockId, sameBlockId, "Equal BlockIds should be equal")
        assertEquals(blockId.hashCode(), sameBlockId.hashCode(), "Equal BlockIds should have the same hash code")
    }
}